### ISSUE TITLE
segmentation fault (core dumped)

### ISSUE THREAD

**Issue by Ivaylo-Korakov** [darvid/python-hyperscan#10]
Hi,

I am trying to use the halting feature of the match_event_handler.
The problem is when I return True I get a `segmentation fault (core dumped)`

My steps: I already have hyperscan installed 
1. `mkvirtualenv test_scan`
2. `pip3 install hyperscan`
3. `python3`
```python
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import hyperscan
>>> 
>>> db = hyperscan.Database()
>>> patterns = (
...     # expression,  id, flags
...     (br'fo+',      0,  0),
...     (br'^foobar$', 1,  hyperscan.HS_FLAG_CASELESS),
...     (br'BAR',      2,  hyperscan.HS_FLAG_CASELESS
...                        | hyperscan.HS_FLAG_SOM_LEFTMOST),
... )
>>> expressions, ids, flags = zip(*patterns)
>>> db.compile(
...     expressions=expressions, ids=ids, elements=len(patterns), flags=flags
... )
>>> print(db.info().decode())
Version: 5.1.1 Features: AVX2 Mode: BLOCK
>>> 
>>> def on_match(*args):
...     print(args)
...     return True
... 
>>> # And now trying to scan 
... 
>>> db.scan(b'foobar', match_event_handler=on_match, context='foo')
(0, 0, 2, 0, 'foo')
[1]    21582 segmentation fault (core dumped)  python3
```
I believe this is not intended. 
The intended result will indeed print out `args` only once and then halt any further scanning

Also, I think I found some inconsistency in the docs/release notes.
In the release notes:
`match_event_handler will now halt scanning if a non-truthy (and
not None) value is returned.`
In the docs:
`The return value determines whether or not Hyperscan should halt scanning. If the match handler returns anything other than None that is truthy, scanning will be halted....`

PS:
Thank you for adding documentation to the package.

**Comment by Ivaylo-Korakov** [comment]
I also tried something else.
Downloaded the newest project from git.
Made the match_event_handle always return 1:
```c
static int match_handler(unsigned int id, unsigned long long from,
                         unsigned long long to, unsigned int flags,
                         void *context) {
  py_scan_callback_ctx *cctx = context;
  PyGILState_STATE gstate;
  gstate = PyGILState_Ensure();
  PyObject *rv = PyObject_CallFunction(cctx->callback, "IIIIO", id,
                                       from, to, flags, cctx->ctx);
  PyGILState_Release(gstate);
  return 1;
}
```

Still got the same issue.

**Comment by darvid** [comment]
should be fixed in the just-released 0.1.1. thanks for reporting, not sure why the tests passed for 0.1.0.
